init.lua

```diff
@@ -72,18 +74,17 @@ function ranks.get_def(rank)
 end
 
 -- [function] Update player privileges
-function ranks.update_privs(player, trigger)
-	if type(player) == "string" then
-		player = minetest.get_player_by_name(player)
+function ranks.update_privs(name, trigger)
+	if type(name) == "string" then
+		player = minetest.get_player_by_name(name)
 	end
 
 	if not player then
 		return
 	end
 
-	local name = player:get_player_name()
-	local rank = ranks.get_rank(player)
-	if rank then
+	local rank = ranks.get_rank(name)
+	if ranks.get_rank(name) ~= nil then
```

Comment:
Line 86 uses the variable `rank` to store the player's rank. There is no need to fetch the rank again on line 87, we need only reference the variable `rank`. The same applies in `update_nametag`, `set_rank` and `remove_rank`.

Reply:
Done!